Compilation error: ‘ZMQ_STREAM_NOTIFY’ was not declared in this scope

I am trying to build it on Ubuntu 16.04 Beta. (to answer your Q in AU). I followed the steps from read-me file.

`make test` raise the following error:
```
$ make test
  CXX      src/libprime_server_la-prime_server.lo
src/prime_server.cpp: In constructor ‘prime_server::client_t::client_t(zmq::context_t&, const string&, const request_function_t&, const collect_function_t&, size_t)’:
src/prime_server.cpp:23:23: error: ‘ZMQ_STREAM_NOTIFY’ was not declared in this scope
     server.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
                       ^
src/prime_server.cpp: In constructor ‘prime_server::server_t<request_container_t, request_info_t>::server_t(zmq::context_t&, const string&, const string&, const string&, bool, size_t)’:
src/prime_server.cpp:91:23: error: ‘ZMQ_STREAM_NOTIFY’ was not declared in this scope
     client.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
                       ^
Makefile:916: recipe for target 'src/libprime_server_la-prime_server.lo' failed
make: *** [src/libprime_server_la-prime_server.lo] Error 1
```

Current dependencies installed:
```
$ apt-cache policy  libcurl4-openssl-dev libzmq3-dev
libcurl4-openssl-dev:
  Installed: 7.47.0-1ubuntu2
  Candidate: 7.47.0-1ubuntu2
  Version table:
 *** 7.47.0-1ubuntu2 500
        500 http://dz.archive.ubuntu.com/ubuntu xenial/main i386 Packages
        100 /var/lib/dpkg/status
libzmq3-dev:
  Installed: 4.1.4-7
  Candidate: 4.1.4-7
  Version table:
 *** 4.1.4-7 500
        500 http://dz.archive.ubuntu.com/ubuntu xenial/universe i386 Packages
        100 /var/lib/dpkg/status
```

--- a/src/prime_server.cpp
+++ b/src/prime_server.cpp
@@ -17,12 +17,6 @@ namespace prime_server {
     int disabled = 0;
     server.setsockopt(ZMQ_SNDHWM, &disabled, sizeof(disabled));
     server.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
-#if ZMQ_VERSION_MAJOR >= 4
-#if ZMQ_VERSION_MINOR >= 1
-    int enabled = 1;
-    server.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-#endif
-#endif
     server.connect(server_endpoint.c_str());
   }
   client_t::~client_t(){}
@@ -85,12 +79,6 @@ namespace prime_server {
     int disabled = 0;
     client.setsockopt(ZMQ_SNDHWM, &disabled, sizeof(disabled));
     client.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
-#if ZMQ_VERSION_MAJOR >= 4
-#if ZMQ_VERSION_MINOR >= 1
-    int enabled = 1;
-    client.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-#endif
-#endif
     client.bind(client_endpoint.c_str());
 
     proxy.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));

--- a/test/zmq.cpp
+++ b/test/zmq.cpp
@@ -20,12 +20,6 @@ namespace {
     int disabled = 0;
     server.setsockopt(ZMQ_SNDHWM, &disabled, sizeof(disabled));
     server.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
-    #if ZMQ_VERSION_MAJOR >= 4
-    #if ZMQ_VERSION_MINOR >= 1
-    int enabled = 1;
-    server.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-    #endif
-    #endif
     server.bind("ipc:///tmp/test_server");
 
     zmq::message_t identity;
@@ -57,12 +51,6 @@ namespace {
     int disabled = 0;
     client.setsockopt(ZMQ_SNDHWM, &disabled, sizeof(disabled));
     client.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
-    #if ZMQ_VERSION_MAJOR >= 4
-    #if ZMQ_VERSION_MINOR >= 1
-    int enabled = 1;
-    client.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-    #endif
-    #endif
     client.connect("ipc:///tmp/test_server");
     #if ZMQ_VERSION_MAJOR >= 4
     #if ZMQ_VERSION_MINOR >= 1
@@ -103,13 +91,6 @@ namespace {
     server.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
     client.setsockopt(ZMQ_SNDHWM, &disabled, sizeof(disabled));
     client.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
-    #if ZMQ_VERSION_MAJOR >= 4
-    #if ZMQ_VERSION_MINOR >= 1
-    int enabled = 1;
-    server.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-    client.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-    #endif
-    #endif
     server.bind("ipc:///tmp/test_server");
     client.connect("ipc:///tmp/test_server");
 
@@ -232,13 +213,6 @@ namespace {
     dealer.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
     router.setsockopt(ZMQ_SNDHWM, &disabled, sizeof(disabled));
     router.setsockopt(ZMQ_RCVHWM, &disabled, sizeof(disabled));
-    #if ZMQ_VERSION_MAJOR >= 4
-    #if ZMQ_VERSION_MINOR >= 1
-    int enabled = 1;
-    client.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-    server.setsockopt(ZMQ_STREAM_NOTIFY, &enabled, sizeof(enabled));
-    #endif
-    #endif
     client.connect("ipc:///tmp/test_server");
     server.bind("ipc:///tmp/test_server");
     dealer.connect("ipc:///tmp/test_router_dealer");

